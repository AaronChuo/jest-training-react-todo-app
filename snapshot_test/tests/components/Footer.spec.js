import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Footer from 'components/Footer';
import {
  SHOW_ALL,
  SHOW_ACTIVE,
  SHOW_COMPLETED,
} from 'constants/TodoFilters';

describe('<Footer />', () => {
  let handleClearCompleted;
  let handleShow;
  let wrapper;

  beforeEach(() => {
    handleClearCompleted = jest.fn();
    handleShow = jest.fn();
    wrapper = shallow(
      <Footer
        completedCount={6}
        activeCount={2}
        filter={SHOW_ALL}
        onClearCompleted={handleClearCompleted}
        onShow={handleShow}
      />
    );
  });

  it('should render initial component', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('should render component without any active todo', () => {
    wrapper.setProps({ activeCount: 0 });
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('should render component with an active todo', () => {
    wrapper.setProps({ activeCount: 1 });
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('should render component with multiple active todos', () => {
    wrapper.setProps({ activeCount: 2 });
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('should render component without any completed todo', () => {
    wrapper.setProps({ completedCount: 0 });
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('should render component with filter SHOW_ACTIVE', () => {
    wrapper.setProps({ filter: SHOW_ACTIVE });
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('should render component with filter SHOW_COMPLETED', () => {
    wrapper.setProps({ filter: SHOW_COMPLETED });
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('simulates click button to filter todos', () => {
    wrapper.find('.filters li a').first().simulate('click');
    expect(handleShow).toHaveBeenCalled();
  });

  it('simulates click button to clear completed todos', () => {
    wrapper.find('.clear-completed').simulate('click');
    expect(handleClearCompleted).toHaveBeenCalled();
  });
});
