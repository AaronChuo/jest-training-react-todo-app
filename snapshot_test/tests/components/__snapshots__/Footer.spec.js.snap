// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render component with an active todo 1`] = `
<footer
  className="footer"
>
  <span
    className="todo-count"
  >
    <strong>
      1
    </strong>
     
    item
     left
  </span>
  <ul
    className="filters"
  >
    <li>
      <a
        className={
          Object {
            "selected": true,
          }
        }
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        All
      </a>
    </li>
    <li>
      <a
        className={
          Object {
            "selected": false,
          }
        }
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        Active
      </a>
    </li>
    <li>
      <a
        className={
          Object {
            "selected": false,
          }
        }
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        Completed
      </a>
    </li>
  </ul>
  <button
    className="clear-completed"
    onClick={[Function]}
  >
    Clear completed
  </button>
</footer>
`;

exports[`<Footer /> should render component with filter SHOW_ACTIVE 1`] = `
<footer
  className="footer"
>
  <span
    className="todo-count"
  >
    <strong>
      2
    </strong>
     
    items
     left
  </span>
  <ul
    className="filters"
  >
    <li>
      <a
        className={
          Object {
            "selected": false,
          }
        }
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        All
      </a>
    </li>
    <li>
      <a
        className={
          Object {
            "selected": true,
          }
        }
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        Active
      </a>
    </li>
    <li>
      <a
        className={
          Object {
            "selected": false,
          }
        }
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        Completed
      </a>
    </li>
  </ul>
  <button
    className="clear-completed"
    onClick={[Function]}
  >
    Clear completed
  </button>
</footer>
`;

exports[`<Footer /> should render component with filter SHOW_COMPLETED 1`] = `
<footer
  className="footer"
>
  <span
    className="todo-count"
  >
    <strong>
      2
    </strong>
     
    items
     left
  </span>
  <ul
    className="filters"
  >
    <li>
      <a
        className={
          Object {
            "selected": false,
          }
        }
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        All
      </a>
    </li>
    <li>
      <a
        className={
          Object {
            "selected": false,
          }
        }
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        Active
      </a>
    </li>
    <li>
      <a
        className={
          Object {
            "selected": true,
          }
        }
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        Completed
      </a>
    </li>
  </ul>
  <button
    className="clear-completed"
    onClick={[Function]}
  >
    Clear completed
  </button>
</footer>
`;

exports[`<Footer /> should render component with multiple active todos 1`] = `
<footer
  className="footer"
>
  <span
    className="todo-count"
  >
    <strong>
      2
    </strong>
     
    items
     left
  </span>
  <ul
    className="filters"
  >
    <li>
      <a
        className={
          Object {
            "selected": true,
          }
        }
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        All
      </a>
    </li>
    <li>
      <a
        className={
          Object {
            "selected": false,
          }
        }
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        Active
      </a>
    </li>
    <li>
      <a
        className={
          Object {
            "selected": false,
          }
        }
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        Completed
      </a>
    </li>
  </ul>
  <button
    className="clear-completed"
    onClick={[Function]}
  >
    Clear completed
  </button>
</footer>
`;

exports[`<Footer /> should render component without any active todo 1`] = `
<footer
  className="footer"
>
  <span
    className="todo-count"
  >
    <strong>
      No
    </strong>
     
    items
     left
  </span>
  <ul
    className="filters"
  >
    <li>
      <a
        className={
          Object {
            "selected": true,
          }
        }
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        All
      </a>
    </li>
    <li>
      <a
        className={
          Object {
            "selected": false,
          }
        }
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        Active
      </a>
    </li>
    <li>
      <a
        className={
          Object {
            "selected": false,
          }
        }
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        Completed
      </a>
    </li>
  </ul>
  <button
    className="clear-completed"
    onClick={[Function]}
  >
    Clear completed
  </button>
</footer>
`;

exports[`<Footer /> should render component without any completed todo 1`] = `
<footer
  className="footer"
>
  <span
    className="todo-count"
  >
    <strong>
      2
    </strong>
     
    items
     left
  </span>
  <ul
    className="filters"
  >
    <li>
      <a
        className={
          Object {
            "selected": true,
          }
        }
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        All
      </a>
    </li>
    <li>
      <a
        className={
          Object {
            "selected": false,
          }
        }
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        Active
      </a>
    </li>
    <li>
      <a
        className={
          Object {
            "selected": false,
          }
        }
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        Completed
      </a>
    </li>
  </ul>
</footer>
`;

exports[`<Footer /> should render initial component 1`] = `
<footer
  className="footer"
>
  <span
    className="todo-count"
  >
    <strong>
      2
    </strong>
     
    items
     left
  </span>
  <ul
    className="filters"
  >
    <li>
      <a
        className={
          Object {
            "selected": true,
          }
        }
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        All
      </a>
    </li>
    <li>
      <a
        className={
          Object {
            "selected": false,
          }
        }
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        Active
      </a>
    </li>
    <li>
      <a
        className={
          Object {
            "selected": false,
          }
        }
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        Completed
      </a>
    </li>
  </ul>
  <button
    className="clear-completed"
    onClick={[Function]}
  >
    Clear completed
  </button>
</footer>
`;
